superset-frontend/src/components/Select/AsyncSelect.tsx

```diff
@@ -523,19 +525,33 @@ const AsyncSelect = forwardRef(
       [ref],
     );
 
+    const pastedTextValue = useMemo(
```

Comment:
1. The name `pastedTextValue` suggests that it's a value, while really it's a function. Should it be called `getPastedTextValue`?
2. `useMemo(() => () => {...}` is the same as `useCallback(() => {...}` - the latter is easier to read when memorising a function